src/Controller/Component/AuthenticationComponent.php

```diff
@@ -373,7 +373,7 @@ public function impersonate(ArrayAccess $impersonated)
         $result = $service->impersonate(
             $controller->getRequest(),
             $controller->getResponse(),
-            $identity,
+            $identity->getOriginalData(),
```

Comment:
Seems like we need to widen the type for `ImpersonationInterface::impersonate()`

Reply:
Fixed this with some 'casting'.